### PR TITLE
Adding new topics on signing, new Cryptography node in left nav

### PR DESCRIPTION
**Pull request by randall-Mysten**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: doc/src/learn/cryptography/sui-intent-signing.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+The `IntentMessage` struct consists of the intent and the serialized data value.
+
+```rust
+pub struct IntentMessage<T> {
+  pub intent: Intent,
+  pub value: T,
+}
+```
+
+To create an intent struct, include the `IntentScope` (what the type of the message is), `IntentVersion` (what version the network supports), `AppId` (what application that the signature refers to).
```
Comment (ronny-mysten) [inline doc/src/learn/cryptography/sui-intent-signing.md:31]:
```suggestion
To create an intent struct, include the `IntentScope` (what the type of the message is), `IntentVersion` (what version the network supports), and `AppId` (what application that the signature refers to).
```

File: doc/src/learn/cryptography/sui-intent-signing.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+```rust
+pub struct Intent {
+  scope: IntentScope,
+  version: IntentVersion,
+  app_id: AppId,
+}
+```
+
+To see detailed definition for each field, see each enum definition [in the source code](https://github.com/MystenLabs/sui/blob/0dc1a38f800fc2d8fabe11477fdef702058cf00d/crates/sui-types/src/intent.rs).
```
Comment (ronny-mysten) [inline doc/src/learn/cryptography/sui-intent-signing.md:41]:
```suggestion
To see a detailed definition for each field, see each enum definition [in the source code](https://github.com/MystenLabs/sui/blob/0dc1a38f800fc2d8fabe11477fdef702058cf00d/crates/sui-types/src/intent.rs).
```

File: doc/src/learn/cryptography/sui-intent-signing.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+The serialization of an `Intent` is a 3-byte array where each field is represented by a byte.
+
+The serialization of an `IntentMessage<T>` is the 3 bytes of the intent concatenated with the BCS serialized message.
+
+## User Signature
+
+To create a user signature, construct a intent message first, and create the signature over the BCS serialized value of the intent message (`intent || message`).
```
Comment (ronny-mysten) [inline doc/src/learn/cryptography/sui-intent-signing.md:49]:
```suggestion
To create a user signature, construct an intent message first, and create the signature over the BCS serialized value of the intent message (`intent || message`).
```

File: doc/src/learn/cryptography/sui-multisig.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
+---
+title: Sui Multi-Signature
+---
+
+Sui supports Multi-Signature (Multisig) transactions. A multisig transaction is one that requires more than one private key to authorize it. This topic demonstrate the workflow to create a multisig transaction in Sui, and then submit it using the Sui CLI against a local network. To learn how to set up a local network, see [Sui Local Network](../build/sui-local-network.md).
```
Comment (ronny-mysten) [inline doc/src/learn/cryptography/sui-multisig.md:5]:
```suggestion
Sui supports multi-signature (multisig) transactions. A multisig transaction is one that requires more than one private key to authorize it. This topic demonstrate the workflow to create a multisig transaction in Sui, and then submit it using the Sui CLI against a local network. To learn how to set up a local network, see [Sui Local Network](../build/sui-local-network.md).
```

File: doc/src/learn/cryptography/sui-multisig.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+```shell
+sui client serialize-transfer-sui --to $$MULTISIG_ADDR --sui-coin-object-id $OBJECT_ID --gas-budget 1000
+
+Raw tx_bytes to execute: $TX_BYTES
+```
+
+## Step 4: Sign the transaction with two keys
+
+Use the following code sample to sign the transaction with two keys in `sui.keystore`. You can do this other tools as long as you serialize it to `flag || sig || pk`.
```
Comment (ronny-mysten) [inline doc/src/learn/cryptography/sui-multisig.md:75]:
```suggestion
Use the following code sample to sign the transaction with two keys in `sui.keystore`. You can do this with other tools as long as you serialize it to `flag || sig || pk`.
```

File: doc/src/learn/cryptography/sui-offline-signing.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+ 
+```shell
+sui client serialize-transfer-sui --to $ADDRESS --sui-coin-object-id $OBJECT_ID --gas-budget 1000
+
+Raw tx_bytes to execute: $TX_BYTES
+```
+
+## Sign the serialized data
+
+You can sign the data using the device and programming language you choose. Sui accepts signatures for pure Ed25519, ECDSA Secp256k1, ECDSA Secp256r1 and native multisig. To learn more about multisig, see [Sui Multi-Signature](sui-multisig.md).
```
Comment (ronny-mysten) [inline doc/src/learn/cryptography/sui-offline-signing.md:25]:
```suggestion
You can sign the data using the device and programming language you choose. Sui accepts signatures for pure ed25519, ECDSA secp256k1, ECDSA secp256r1 and native multisig. To learn more about multisig, see [Sui Multi-Signature](sui-multisig.md).
```

File: doc/src/learn/cryptography/sui-offline-signing.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+The signature is committed to an intent message of the transaction data. To learn how to construct an intent message, see [Sui Intent signing](sui-intent_signing.md). 
+
+Before you pass it in to the signing API, you must first hash the intent message to 32 bytes using with Blake2b. Depending on the signing scheme, Sui performs an additional hashing is internally to the signing API. For ECDSA Secp256k1 and Secp256r1, you must use SHA-2 SHA256 as the internal hash function. For pure Ed25519, you must use SHA-512.
```
Comment (ronny-mysten) [inline doc/src/learn/cryptography/sui-offline-signing.md:29]:
```suggestion
Before you pass it in to the signing API, you must first hash the intent message to 32 bytes using with BLAKE2b. Depending on the signing scheme, Sui performs an additional hashing is internally to the signing API. For ECDSA secp256k1 and secp256r1, you must use SHA-2 SHA256 as the internal hash function. For pure ed25519, you must use SHA-512. 
```
"Sui performs an additional hashing is internally to the signing API": Not sure how to fix?

File: doc/src/learn/cryptography/sui-offline-signing.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+You can sign the data using the device and programming language you choose. Sui accepts signatures for pure Ed25519, ECDSA Secp256k1, ECDSA Secp256r1 and native multisig. To learn more about multisig, see [Sui Multi-Signature](sui-multisig.md).
+
+The signature is committed to an intent message of the transaction data. To learn how to construct an intent message, see [Sui Intent signing](sui-intent_signing.md). 
+
+Before you pass it in to the signing API, you must first hash the intent message to 32 bytes using with Blake2b. Depending on the signing scheme, Sui performs an additional hashing is internally to the signing API. For ECDSA Secp256k1 and Secp256r1, you must use SHA-2 SHA256 as the internal hash function. For pure Ed25519, you must use SHA-512. 
+
+Additional signature requirements:
+
+An accepted ECDSA Secp256k1 and Secp256r1 signature must follow:
```
Comment (ronny-mysten) [inline doc/src/learn/cryptography/sui-offline-signing.md:33]:
```suggestion
An accepted ECDSA secp256k1 and secp256r1 signature must follow:
```

File: doc/src/learn/cryptography/sui-offline-signing.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+ 1. The signature must be of length 64 bytes in the form of `[r, s]` where the first 32 bytes are `r`, the second 32 bytes are `s`.
+ 1. The `r` value can be between 0x1 and 0xFFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFE BAAEDCE6 AF48A03B BFD25E8C D0364140 (inclusive).
+ 1. The `s` value must be in the lower half of the curve order. If the signature is too high, please convert it to a lower `s` according to [BIP-0062](https://github.com/bitcoin/bips/blob/master/bip-0062.mediawiki#low-s-values-in-signatures) with the corresponding curve orders using `order - s`. For Secp256k1, the curve order is `0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141`. For Secp256r1, the curve order is `0xFFFFFFFF00000000FFFFFFFFFFFFFFFFBCE6FAADA7179E84F3B9CAC2FC632551` defined in [Standards for Efficient Cryptography](https://secg.org/SEC2-Ver-1.0.pdf).
```
Comment (ronny-mysten) [inline doc/src/learn/cryptography/sui-offline-signing.md:37]:
```suggestion
 1. The `s` value must be in the lower half of the curve order. If the signature is too high, convert it to a lower `s` according to [BIP-0062](https://github.com/bitcoin/bips/blob/master/bip-0062.mediawiki#low-s-values-in-signatures) with the corresponding curve orders using `order - s`. For secp256k1, the curve order is `0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141`. For secp256r1, the curve order is `0xFFFFFFFF00000000FFFFFFFFFFFFFFFFBCE6FAADA7179E84F3B9CAC2FC632551` defined in [Standards for Efficient Cryptography](https://secg.org/SEC2-Ver-1.0.pdf).
```

File: doc/src/learn/cryptography/sui-offline-signing.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+ 1. Ideally, the signature must be generated with deterministic nonce according to [RFC6979](https://www.rfc-editor.org/rfc/rfc6979).
+
+An accepted pure Ed25519 signature follows:
```
Comment (ronny-mysten) [inline doc/src/learn/cryptography/sui-offline-signing.md:40]:
```suggestion
An accepted pure ed25519 signature follows:
```

File: doc/src/learn/cryptography/sui-offline-signing.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+
+Signer address: $ADDRESS
+Raw tx_bytes to execute: $TX_BYTES
+Intent: Intent { scope: TransactionData, version: V0, app_id: Sui }
+Raw intent message: $INTENT_MSG
+Digest to sign: $DIGEST
+Serialized signature (`flag || sig || pk` in Base64): $SERIALIZED_SIG
+```
+
+To ensure the signature produced offline matches with Sui's validity rules for testing purpose, you can import the mnemonics to `sui.keystore` using `sui keytool import`. You can then sign with it using `sui keytool sign` and then compare the signature results. Additionally, you can find test vectors in `~/sui/sdk/typescript/test/e2e/raw-signer.test.ts`.
```
Comment (ronny-mysten) [inline doc/src/learn/cryptography/sui-offline-signing.md:57]:
```suggestion
To ensure the signature produced offline matches with Sui's validity rules for testing purposes, you can import the mnemonics to `sui.keystore` using `sui keytool import`. You can then sign with it using `sui keytool sign` and then compare the signature results. Additionally, you can find test vectors in `~/sui/sdk/typescript/test/e2e/raw-signer.test.ts`. 
```

**Review by ronny-mysten — APPROVED** [review]

File: doc/src/learn/cryptography/sui-multisig.md [hunk 12]
```diff
@@ -0,0 +1,101 @@
+---
+title: Sui Multi-Signature
+---
+
+Sui supports multi-signature (multisig) transactions. A multisig transaction is one that requires more than one private key to authorize it. This topic demonstrate the workflow to create a multisig transaction in Sui, and then submit it using the Sui CLI against a local network. To learn how to set up a local network, see [Sui Local Network](../build/sui-local-network.md).
```
Comment (kchalkias) [inline doc/src/learn/cryptography/sui-multisig.md:5]: Let's add that we support k out of n where n <= 10

File: doc/src/learn/cryptography/sui-offline-signing.md (hunk 7 above)
Comment (kchalkias) [reply doc/src/learn/cryptography/sui-offline-signing.md:29]: Revise to: "To be compatible with existing standards and hardware secure modules (HSMs), the signing algorithms perform additional hashing internally."

File: doc/src/learn/cryptography/sui-offline-signing.md [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+Raw tx_bytes to execute: $TX_BYTES
+```
+
+## Sign the serialized data
+
+You can sign the data using the device and programming language you choose. Sui accepts signatures for pure ed25519, ECDSA secp256k1, ECDSA secp256r1 and native multisig. To learn more about multisig, see [Sui Multi-Signature](sui-multisig.md).
+
+The signature is committed to an intent message of the transaction data. To learn how to construct an intent message, see [Sui Intent signing](sui-intent_signing.md). 
+
+Before you pass it in to the signing API, you must first hash the intent message to 32 bytes using with Blake2b. Depending on the signing scheme, Sui performs an additional hashing is internally to the signing API. For ECDSA Secp256k1 and Secp256r1, you must use SHA-2 SHA256 as the internal hash function. For pure Ed25519, you must use SHA-512.
```
Comment (kchalkias) [inline doc/src/learn/cryptography/sui-offline-signing.md:29]: > hash the intent message to 32 bytes using ~with~ BLAKE2b.

File: doc/src/learn/cryptography/sui-multisig.md (hunk 12 above)
Comment (randall-Mysten) [reply doc/src/learn/cryptography/sui-multisig.md:5]: added

File: doc/src/learn/cryptography/sui-offline-signing.md (hunk 13 above)
Comment (randall-Mysten) [reply doc/src/learn/cryptography/sui-offline-signing.md:29]: added

**Review by kchalkias — APPROVED** [review]
LGTM, Linter's reported errors are false positives, we're good to go.

**Pull request merged**